### DISCUSSION TITLE
Module detection in multilayer network with different type of nodes in each layer

### DISCUSSION THREAD

**Discussion by InaMariaDeutschmann** [mapequation/infomap#281] [General]
Dear Infomap devs,

I would like to use your tool to find modules in a specific multilayer network. From your toy example (https://www.mapequation.org/infomap/#InputMultilayerFull), I see that it is possible to connect different physical nodes through state nodes belonging to two different layers. For example, node 1 in layer 1 connects to node 2 and 3 in layer 2.

```
# A multilayer network using explicit format
# Figure 3
*Vertices 5
# node_id name
1 "i" 
2 "j" 
3 "k" 
4 "l" 
5 "m" 
*Multilayer
# layer_id node_id layer_id node_id weight
# intra
1 1 1 4 0.8 
1 4 1 1 1 
1 1 1 5 0.8 
1 5 1 1 1 
1 4 1 5 1 
1 5 1 4 1 
2 1 2 2 0.8 
2 2 2 1 1 
2 1 2 3 0.8 
2 3 2 1 1 
2 2 2 3 1 
2 3 2 2 1 
# inter
1 1 2 2 0.2 
1 1 2 3 0.2 
2 1 1 4 0.2 
2 1 1 5 0.2 
```

Equation (1) in (https://doi.org/10.1103/PhysRevX.5.011027) shows that only intralayer edges between state nodes of a physical node are considered (s_i^beta) but not intralayer edges of two state nodes belonging to two physical nodes.

There are no such edges in our network because we do not allow a node to be in more than one layer. A given node i has directed edges within a layer, and can also have an edge to state node j that is in the another layer, and j is different from i, i.e., the state nodes i and j belong to two different physical nodes. Thus, is it possible to apply your program to such a multilayer network in which each layer consists of different types of nodes? If I understood correctly, then in our network, every state node would equal the physical node, right? How are the transition probabilities (equation 1) adjusted that the flows use to explore the multi-layer network for module detection?

Thank you,
Ina

**Comment by danieledler** [comment]
Hi Ina,

Thanks for your question. If the layers contain disjoint sets of nodes, there is as you said only one state node per physical node. The model in Eq. 1 you refer to is for the [intra/inter format](https://www.mapequation.org/infomap/#InputMultilayerIntraInter), where inter-layer links are constrained to the same physical node. So this model doesn't support your use case.

However, in the full multilayer format you used in your example, where you explicitly define all multilayer links, there is no model constraining or altering what you put in. The transition probabilities in this state network are not calculated according to Eq. 1 but as for ordinary PageRank: the probability of following a link from node a to b, irrespective of layer, is simply the weight of that link divided by the sum of weights on all links going out from node a.

Does this answer your questions?
